src/core/layertree/qgslayertreecustomnode.cpp

```diff
@@ -0,0 +1,95 @@
+/***************************************************************************
+                       qgslayertreecustomnode.cpp
+                       --------------------------
+    begin                : July 2025
+    copyright            : (C) 2025 by Germán Carrillo
+    email                : german at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayertreecustomnode.h"
+#include "moc_qgslayertreecustomnode.cpp"
+#include "qgslayertreeutils.h"
+
+
+QgsLayerTreeCustomNode::QgsLayerTreeCustomNode( const QString &nodeId, const QString &nodeName, bool checked )
+  : QgsLayerTreeNode( NodeCustom, checked )
+  , mId( nodeId )
+  , mName( nodeName.isEmpty() ? nodeId : nodeName )
+{
+}
+
+QgsLayerTreeCustomNode::QgsLayerTreeCustomNode( const QgsLayerTreeCustomNode &other )
+  : QgsLayerTreeNode( other )
+  , mId( other.mId )
+  , mName( other.mName )
+{
+}
+
+QString QgsLayerTreeCustomNode::name() const
+{
+  return mName;
+}
+
+void QgsLayerTreeCustomNode::setName( const QString &name )
+{
+  if ( mName == name )
+    return;
+
+  mName = name;
+  emit nameChanged( this, name );
+}
+
+QgsLayerTreeCustomNode *QgsLayerTreeCustomNode::readXml( const QDomElement &element, const QgsReadWriteContext & ) // cppcheck-suppress duplInheritedMember
+{
+  if ( element.tagName() != QLatin1String( "layer-tree-custom-node" ) )
+    return nullptr;
+
+  const QString nodeId = element.attribute( QStringLiteral( "id" ) );
+  const QString name =  element.attribute( QStringLiteral( "name" ) );
+  bool checked = QgsLayerTreeUtils::checkStateFromXml( element.attribute( QStringLiteral( "checked" ) ) ) != Qt::Unchecked;
+  bool isExpanded = ( element.attribute( QStringLiteral( "expanded" ), QStringLiteral( "1" ) ) == QLatin1String( "1" ) );
```

Comment:
does the expanded state makes sense for custom node? is this usable?

Reply:
Good point. Let's remove it now and let other devs to add it if ever needed. In any case, it comes from the base class, so it'll always be available.